src/data/automation_i18n.ts

```diff
@@ -199,126 +199,91 @@ const tryDescribeTrigger = (
 
   // State Trigger
   if (trigger.platform === "state") {
-    let base = "When";
     const entities: string[] = [];
     const states = hass.states;
 
+    let attribute = "";
     if (trigger.attribute) {
       const stateObj = Array.isArray(trigger.entity_id)
         ? hass.states[trigger.entity_id[0]]
         : hass.states[trigger.entity_id];
-      base += ` ${computeAttributeNameDisplay(
+      attribute = computeAttributeNameDisplay(
         hass.localize,
         stateObj,
         hass.entities,
         trigger.attribute
-      )} of`;
-    }
-
-    if (Array.isArray(trigger.entity_id)) {
-      for (const entity of trigger.entity_id.values()) {
-        if (states[entity]) {
-          entities.push(computeStateName(states[entity]) || entity);
-        }
-      }
-    } else if (trigger.entity_id) {
-      entities.push(
-        states[trigger.entity_id]
-          ? computeStateName(states[trigger.entity_id])
-          : trigger.entity_id
       );
     }
 
-    if (entities.length === 0) {
-      // no entity_id or empty array
-      entities.push("something");
+    let entityArray: string[] = [];
+    if (!Array.isArray(trigger.entity_id)) {
+      entityArray = [trigger.entity_id];
+    } else {
+      entityArray = trigger.entity_id;
+    }
+    for (const entity of entityArray.values()) {
+      if (states[entity]) {
+        entities.push(computeStateName(states[entity]) || entity);
+      }
     }
 
-    base += ` ${entities} changes`;
+    const stateObj = hass.states[trigger.entity_id[0]];
```

Comment:
`trigger.entity_id` can be a string, so then this would only ge the first char of the entity id.
```suggestion
    const stateObj = hass.states[entityArray[0]];
```